libc/include/sys/rtinit.h

```diff
@@ -32,8 +32,6 @@ extern struct _rt_init _Start_YI;
 extern struct _rt_init _End_YI;
 
 /* constructor/destructor functions called before main and after exit */
-#pragma aux __InitRtns modify [ bx cx dx si di ]
-#pragma aux __FiniRtns modify [ bx cx dx si di ]
 extern void __InitRtns(void);
```

Comment:
Isn't the pragma aux required to tell OWC that the called function doesn't save bx/cx/dx/si/di? We don't want any extra code generated from the calling function (premain) to save registers if it thinks they may be modified by the constructor/destructors. I am trying to keep premain code very efficient and small.

Or is this pragma completely unneeded in this case?

Reply:
You broked OW compiler logic by these #pragmas that it do ineficient things. I checked code and now without such strange pragmas is code little bit smaller because compiler replace some far calls by jumps. OW compiler propagate no return on the path that if you broke this path by your pragma then it is less eficient.
You can look on generated code. Now I removed all these strange pragmas everywhere and use __declspec(aborts) for noreturn macro, it is all. Compiler do all work for me and it works anywhere without strange pragmas and for any calling convention because compiler know all details about each function.
__declspec(aborts) switch on optimization to remove useless frames, and it requires also optimization on called function if it is no return, because if you remove frame then you need correct stack offset calculation in called function etc. It is what OW compiler do optimize both sides and then it is more efficien then C standard _Noreturn modifier which can only break flow control but cannot remove frame.